File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        // Map to store sorted string -> list of anagrams
        Map<String, List<String>> map = new HashMap<>();
        
        for (String str : strs) {
            // Sort the string to use as key
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String sorted = new String(chars);
            
            // Add to the appropriate group
            if (!map.containsKey(sorted)) {
                map.put(sorted, new ArrayList<>());
            }
            map.get(sorted).add(str);
        }
        
        // Return all groups
        return new ArrayList<>(map.values());
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        for s in strs:
            key = ''.join(sorted(s))
            if key in groups:
                groups[key].append(s)
            else:
                groups[key] = [s]
        return list(groups.values())